use JIT'ed fp16 SLS

jspark1105 opened the pull request:
Summary: Use JIT'ed fp16 SLS in D19477209 from Caffe2 operators

Test Plan: CI

Differential Revision: D19477208

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit e8d67103:


* **5/5** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=e8d67103384162103a924e90325d5a58d9deb464).

### :detective: 3 new failures recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4303778) pytorch_linux_xenial_py2_7_9_build (1/3)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=88746755) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=88746755))
<details>
<summary>
<code>Jan 21 06:38:42 /var/lib/jenkins/workspace/build/lib/libtorch_cpu.so: undefined reference to `fbgemm::EmbeddingSpMDMKernelSignature<unsigned short, int>::Type fbgemm::GenerateEmbeddingSpMDM<unsigned short, int>(long, bool, bool, int, bool)'</code>
</summary>

```
Jan 21 06:38:42 make[1]: *** Waiting for unfinished jobs.... 
Jan 21 06:38:42 CMakeFiles/Makefile2:3606: recipe for target 'caffe2/CMakeFiles/pattern_net_transform_test.dir/all' failed 
Jan 21 06:38:42 /var/lib/jenkins/workspace/build/lib/libtorch_cpu.so: undefined reference to `fbgemm::EmbeddingSpMDMKernelSignature<unsigned short, long>::Type fbgemm::GenerateEmbeddingSpMDM<unsigned short, long>(long, bool, bool, int, bool)' 
Jan 21 06:38:42 /var/lib/jenkins/workspace/build/lib/libtorch_cpu.so: undefined reference to `fbgemm::EmbeddingSpMDMKernelSignature<unsigned short, int>::Type fbgemm::GenerateEmbeddingSpMDM<unsigned short, int>(long, bool, bool, int, bool)' 
Jan 21 06:38:42 collect2: error: ld returned 1 exit status 
Jan 21 06:38:42 make[2]: *** [bin/depthwise3x3_conv_op_test] Error 1 
Jan 21 06:38:42 caffe2/CMakeFiles/depthwise3x3_conv_op_test.dir/build.make:105: recipe for target 'bin/depthwise3x3_conv_op_test' failed 
Jan 21 06:38:42 make[1]: *** [caffe2/CMakeFiles/depthwise3x3_conv_op_test.dir/all] Error 2 
Jan 21 06:38:42 CMakeFiles/Makefile2:3649: recipe for target 'caffe2/CMakeFiles/depthwise3x3_conv_op_test.dir/all' failed 
Jan 21 06:38:42 /var/lib/jenkins/workspace/build/lib/libtorch_cpu.so: undefined reference to `fbgemm::EmbeddingSpMDMKernelSignature<unsigned short, long>::Type fbgemm::GenerateEmbeddingSpMDM<unsigned short, long>(long, bool, bool, int, bool)' 
Jan 21 06:38:42 /var/lib/jenkins/workspace/build/lib/libtorch_cpu.so: undefined reference to `fbgemm::EmbeddingSpMDMKernelSignature<unsigned short, int>::Type fbgemm::GenerateEmbeddingSpMDM<unsigned short, int>(long, bool, bool, int, bool)' 
Jan 21 06:38:42 collect2: error: ld returned 1 exit status 
Jan 21 06:38:42 make[2]: *** [bin/conv_to_nnpack_transform_test] Error 1 
Jan 21 06:38:42 caffe2/CMakeFiles/conv_to_nnpack_transform_test.dir/build.make:105: recipe for target 'bin/conv_to_nnpack_transform_test' failed 
Jan 21 06:38:42 make[1]: *** [caffe2/CMakeFiles/conv_to_nnpack_transform_test.dir/all] Error 2 
Jan 21 06:38:42 CMakeFiles/Makefile2:3563: recipe for target 'caffe2/CMakeFiles/conv_to_nnpack_transform_test.dir/all' failed 
Jan 21 06:38:42 Makefile:138: recipe for target 'all' failed 
Jan 21 06:38:42 make: *** [all] Error 2 
Jan 21 06:38:42 Traceback (most recent call last): 
Jan 21 06:38:42   File "setup.py", line 737, in <module> 
Jan 21 06:38:42 Building wheel torch-1.5.0a0+e8d6710 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4303787) pytorch_linux_xenial_py3_6_gcc5_4_build (2/3)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=88746801) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=88746801))
<details>
<summary>
<code>Jan 21 06:39:06 /var/lib/jenkins/workspace/build/lib/libtorch_cpu.so: undefined reference to `fbgemm::EmbeddingSpMDMKernelSignature<unsigned short, long>::Type fbgemm::GenerateEmbeddingSpMDM<unsigned short, long>(long, bool, bool, int, bool)'</code>
</summary>

```
Jan 21 06:39:06 make[1]: *** [caffe2/CMakeFiles/depthwise3x3_conv_op_test.dir/all] Error 2 
Jan 21 06:39:06 make[1]: *** Waiting for unfinished jobs.... 
Jan 21 06:39:06 /var/lib/jenkins/workspace/build/lib/libtorch_cpu.so: undefined reference to `fbgemm::EmbeddingSpMDMKernelSignature<unsigned short, int>::Type fbgemm::GenerateEmbeddingSpMDM<unsigned short, int>(long, bool, bool, int, bool)' 
Jan 21 06:39:06 /var/lib/jenkins/workspace/build/lib/libtorch_cpu.so: undefined reference to `fbgemm::EmbeddingSpMDMKernelSignature<unsigned short, long>::Type fbgemm::GenerateEmbeddingSpMDM<unsigned short, long>(long, bool, bool, int, bool)' 
Jan 21 06:39:06 collect2: error: ld returned 1 exit status 
Jan 21 06:39:06 caffe2/CMakeFiles/conv_to_nnpack_transform_test.dir/build.make:105: recipe for target 'bin/conv_to_nnpack_transform_test' failed 
Jan 21 06:39:06 make[2]: *** [bin/conv_to_nnpack_transform_test] Error 1 
Jan 21 06:39:06 CMakeFiles/Makefile2:3563: recipe for target 'caffe2/CMakeFiles/conv_to_nnpack_transform_test.dir/all' failed 
Jan 21 06:39:06 make[1]: *** [caffe2/CMakeFiles/conv_to_nnpack_transform_test.dir/all] Error 2 
Jan 21 06:39:06 /var/lib/jenkins/workspace/build/lib/libtorch_cpu.so: undefined reference to `fbgemm::EmbeddingSpMDMKernelSignature<unsigned short, int>::Type fbgemm::GenerateEmbeddingSpMDM<unsigned short, int>(long, bool, bool, int, bool)' 
Jan 21 06:39:06 /var/lib/jenkins/workspace/build/lib/libtorch_cpu.so: undefined reference to `fbgemm::EmbeddingSpMDMKernelSignature<unsigned short, long>::Type fbgemm::GenerateEmbeddingSpMDM<unsigned short, long>(long, bool, bool, int, bool)' 
Jan 21 06:39:06 collect2: error: ld returned 1 exit status 
Jan 21 06:39:06 caffe2/CMakeFiles/pattern_net_transform_test.dir/build.make:105: recipe for target 'bin/pattern_net_transform_test' failed 
Jan 21 06:39:06 make[2]: *** [bin/pattern_net_transform_test] Error 1 
Jan 21 06:39:06 CMakeFiles/Makefile2:3606: recipe for target 'caffe2/CMakeFiles/pattern_net_transform_test.dir/all' failed 
Jan 21 06:39:06 make[1]: *** [caffe2/CMakeFiles/pattern_net_transform_test.dir/all] Error 2 
Jan 21 06:39:06 make: *** [all] Error 2 
Jan 21 06:39:06 Makefile:138: recipe for target 'all' failed 
Jan 21 06:39:06 -- Building version 1.5.0a0+e8d6710 
ld -DBUILD_PYTHON=True -DBUILD_TEST=True -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/var/lib/jenkins/workspace/torch -DCMAKE_PREFIX_PATH=/opt/conda/lib/python3.6/site-packages -DNUMPY_INCLUDE_DIR=/opt/conda/lib/python3.6/site-packages/numpy/core/include -DPYTHON_EXECUTABLE=/opt/conda/bin/python -DPYTHON_INCLUDE_DIR=/opt/conda/include/python3.6m -DPYTHON_LIBRARY=/opt/conda/lib/libpython3.6m.so.1.0 -DTORCH_BUILD_VERSION=1.5.0a0+e8d6710 -DUSE_NUMPY=True -DWERROR=1 /var/lib/jenkins/workspace 
Jan 21 06:39:06 cmake --build . --target install --config Release -- -j 3 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4303789) pytorch_linux_xenial_py3_clang5_asan_build (3/3)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=88746858) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=88746858))
<details>
<summary>
<code>Jan 21 06:39:17 ../lib/libtorch_cpu.so: undefined reference to `fbgemm::EmbeddingSpMDMKernelSignature<unsigned short, long>::Type fbgemm::GenerateEmbeddingSpMDM<unsigned short, long>(long, bool, bool, int, bool)'</code>
</summary>

```
Jan 21 06:39:17 make[1]: *** [caffe2/CMakeFiles/conv_to_nnpack_transform_test.dir/all] Error 2 
Jan 21 06:39:17 make[1]: *** Waiting for unfinished jobs.... 
Jan 21 06:39:17 ../lib/libtorch_cpu.so: undefined reference to `fbgemm::EmbeddingSpMDMKernelSignature<unsigned short, int>::Type fbgemm::GenerateEmbeddingSpMDM<unsigned short, int>(long, bool, bool, int, bool)' 
Jan 21 06:39:17 ../lib/libtorch_cpu.so: undefined reference to `fbgemm::EmbeddingSpMDMKernelSignature<unsigned short, long>::Type fbgemm::GenerateEmbeddingSpMDM<unsigned short, long>(long, bool, bool, int, bool)' 
Jan 21 06:39:17 clang: error: linker command failed with exit code 1 (use -v to see invocation) 
Jan 21 06:39:17 caffe2/CMakeFiles/pattern_net_transform_test.dir/build.make:107: recipe for target 'bin/pattern_net_transform_test' failed 
Jan 21 06:39:17 make[2]: *** [bin/pattern_net_transform_test] Error 1 
Jan 21 06:39:17 CMakeFiles/Makefile2:3489: recipe for target 'caffe2/CMakeFiles/pattern_net_transform_test.dir/all' failed 
Jan 21 06:39:17 make[1]: *** [caffe2/CMakeFiles/pattern_net_transform_test.dir/all] Error 2 
Jan 21 06:39:17 ../lib/libtorch_cpu.so: undefined reference to `fbgemm::EmbeddingSpMDMKernelSignature<unsigned short, int>::Type fbgemm::GenerateEmbeddingSpMDM<unsigned short, int>(long, bool, bool, int, bool)' 
Jan 21 06:39:17 ../lib/libtorch_cpu.so: undefined reference to `fbgemm::EmbeddingSpMDMKernelSignature<unsigned short, long>::Type fbgemm::GenerateEmbeddingSpMDM<unsigned short, long>(long, bool, bool, int, bool)' 
Jan 21 06:39:17 clang: error: linker command failed with exit code 1 (use -v to see invocation) 
Jan 21 06:39:17 caffe2/CMakeFiles/depthwise3x3_conv_op_test.dir/build.make:107: recipe for target 'bin/depthwise3x3_conv_op_test' failed 
Jan 21 06:39:17 make[2]: *** [bin/depthwise3x3_conv_op_test] Error 1 
Jan 21 06:39:17 CMakeFiles/Makefile2:3531: recipe for target 'caffe2/CMakeFiles/depthwise3x3_conv_op_test.dir/all' failed 
Jan 21 06:39:17 make[1]: *** [caffe2/CMakeFiles/depthwise3x3_conv_op_test.dir/all] Error 2 
Jan 21 06:39:17 make: *** [all] Error 2 
Jan 21 06:39:17 Makefile:138: recipe for target 'all' failed 
Jan 21 06:39:17 -- Building version 1.5.0a0+e8d6710 
_TEST=True -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/var/lib/jenkins/workspace/torch -DCMAKE_PREFIX_PATH=/opt/conda/lib/python3.6/site-packages -DNUMPY_INCLUDE_DIR=/opt/conda/lib/python3.6/site-packages/numpy/core/include -DPYTHON_EXECUTABLE=/opt/conda/bin/python -DPYTHON_INCLUDE_DIR=/opt/conda/include/python3.6m -DPYTHON_LIBRARY=/opt/conda/lib/libpython3.6m.so.1.0 -DTORCH_BUILD_VERSION=1.5.0a0+e8d6710 -DUSE_ASAN=1 -DUSE_CUDA=0 -DUSE_MKLDNN=0 -DUSE_NUMPY=True /var/lib/jenkins/workspace 
Jan 21 06:39:17 cmake --build . --target install --config Release -- -j 3 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

jspark1105 committed:
[caffe2] use JIT'ed fp16 SLS (#32432)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32432

Use JIT'ed fp16 SLS in D19477209 from Caffe2 operators

Test Plan: CI

Differential Revision: D19477208

fbshipit-source-id: 94ef5d96d67fb61ab2056caefb9056e97fda1a71
